autorest/codegen/serializers/builder_serializer.py

```diff
@@ -33,9 +33,12 @@
 OrderedSet = Dict[T, None]
 
 
+def _improve_json_string(template_representation: str) -> Any:
+    return template_representation.replace('",', '').replace('#%#', '",').replace(',"', '').replace('."', '.')
+
 def _json_dumps_template(template_representation: Any) -> Any:
     # only for template use, since it wraps everything in strings
-    return json.dumps(template_representation, sort_keys=True, indent=4)
+    return _improve_json_string(json.dumps(template_representation, sort_keys=True, indent=4))
```

Comment:
![image](https://user-images.githubusercontent.com/70930885/131625374-7cab1d29-c500-43e5-9ef5-bc59ecd913ff.png)
 
@iscai-msft No matter how I handle the string before `json.dumps()`, we need to transform it to the format that we want. Do you have any good idea?